File: src/main/python/g0201_0300/s0287_find_the_duplicate_number/Solution.py
# #Medium #Top_100_Liked_Questions #Array #Binary_Search #Two_Pointers #Bit_Manipulation
# #Binary_Search_II_Day_5 #Big_O_Time_O(n)_Space_O(n)
# #2024_06_08_Time_453_ms_(81.73%)_Space_31.1_MB_(24.78%)

class Solution:
    def findDuplicate(self, nums: List[int]) -> int:
        arr = [0] * (len(nums) + 1)
        for num in nums:
            arr[num] += 1
            if arr[num] == 2:
                return num
        return 0

File: src/main/dart/g0201_0300/s0287_find_the_duplicate_number/Solution.dart
// #Medium #Top_100_Liked_Questions #Array #Binary_Search #Two_Pointers #Bit_Manipulation
// #Binary_Search_II_Day_5 #Big_O_Time_O(n)_Space_O(n)
// #2024_10_09_Time_376_ms_(92.11%)_Space_179.8_MB_(44.74%)

class Solution {
  int findDuplicate(List<int> nums) {
    List<int> arr = List.filled(nums.length + 1, 0);

    for (int num in nums) {
      arr[num] += 1;
      if (arr[num] == 2) {
        return num;
      }
    }

    return 0;
  }
}
